tests/gis_tests/gdal_tests/test_raster.py

```diff
@@ -318,6 +318,19 @@ def test_band_data(self):
         self.assertEqual(self.band.datatype(), 1)
         self.assertEqual(self.band.datatype(as_string=True), 'GDT_Byte')
         self.assertEqual(self.band.nodata_value, 15)
+        if numpy:
+            data = self.band.data()
+            # Check that the 42 first values of the column 42 match what's
+            # inside the file raster.tif
+            correct_list = [4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
+                            4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 3, 3, 9, 9,
+                            9, 4, 4, 4, 3, 3, 4, 4, 4, 4, 4, 4, 4, 4]
+            for x in range(0, 42):
```

Comment:
Since here numpy is available, you could avoid this loop by making the correct list a numpy array and to use `numpy.testing.assert_equal`.